pool/paymentmgr_test.go

```diff
@@ -0,0 +1,762 @@
+package pool
+
+import (
+	"fmt"
+	"math/big"
+	"testing"
+	"time"
+
+	bolt "github.com/coreos/bbolt"
+	"github.com/decred/dcrd/chaincfg/v2"
+	"github.com/decred/dcrd/dcrutil/v2"
+)
+
+func testPaymentMgr(t *testing.T, db *bolt.DB) {
+	minPayment, err := dcrutil.NewAmount(2.0)
+	if err != nil {
+		t.Fatalf("[NewAmount] unexpected error: %v", err)
+	}
+	maxTxFeeReserve, err := dcrutil.NewAmount(0.1)
+	if err != nil {
+		t.Fatalf("[NewAmount] unexpected error: %v", err)
+	}
+	activeNet := chaincfg.SimNetParams()
+	activeNet.CoinbaseMaturity = 0 // allow immediately mature coinbases, for testing.
```

Comment:
I've just noticed this here after writing my other comment which states that CoinbaseMaturity is never 0.

Do we really want to be testing with this value if the production code never uses it? Seems like it could introduce bugs because we are testing code which is never run in production, and we have code which is always run in production but never tested.

Reply:
noted, I'm going to take another look at it.

Reply:
good catch, removed it and updated the tests accordingly.